docs/cassandra.md

```diff
@@ -0,0 +1,64 @@
+| `collect_jvm_metrics` | true               | Should the set of support [JVM metrics](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/docs/jvm.md#metrics) also be collected |
+| `username`            | not set by default | The configured username if JMX is configured to require authentication. |
+| `password`            | not set by default | The configured password if JMX is configured to require authentication. |
+| `collection_interval` | `60s`              | A [time.Duration](https://pkg.go.dev/time#ParseDuration) value, such as `30s` or `5m`. |
+
+Example Configuration:
+
+```yaml
+metrics:
+  receivers:
+    cassandra_metrics:
+      type: cassandra
+      endpoint: localhost:7199
+      collection_interval: 30s
+  service:
+    pipelines:
+      cassandra_pipeline:
+        receivers:
+          - cassandra_metrics
+```
+
+## Metrics
+In addition to Cassandra specific metrics, by default Cassandra will also report [JVM metrics](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/docs/jvm.md#metrics)
+
+| Metric                                                                          | Data Type | Unit        | Labels | Description |
+| ---                                                                             | ---       | ---         | ---    | ---         | 
+| workload.googleapis.com/cassandra.client.request.read.latency.50p               | gauge     | µs          |        | Standard read request latency - 50th percentile |
+| workload.googleapis.com/cassandra.client.request.read.latency.99p               | gauge     | µs          |        | Standard read request latency - 99th percentile |
+| workload.googleapis.com/cassandra.client.request.read.latency.count             | sum       | µs          |        | Total standard read request latency |
+| workload.googleapis.com/cassandra.client.request.read.latency.max               | gauge     | µs          |        | Maximum standard read request latency |
+| workload.googleapis.com/cassandra.client.request.read.timeout.count             | sum       | 1           |        | Number of standard read request timeouts encountered |
+| workload.googleapis.com/cassandra.client.request.read.unavailable.count         | sum       | 1           |        | Number of standard read request unavailable exceptions encountered |
+| workload.googleapis.com/cassandra.client.request.write.latency.50p              | gauge     | µs          |        | Standard write request latency - 50th percentile |
+| workload.googleapis.com/cassandra.client.request.write.latency.99p              | gauge     | µs          |        | Standard write request latency - 99th percentile |
+| workload.googleapis.com/cassandra.client.request.write.latency.count            | sum       | µs          |        | Total standard write request latency |
```

Comment:
It's really weird for the write request count to have `.latency.` in the name, but I suppose that's an unfortunate consequence of how these are exposed.

Reply:
I believe this metric represents the total amount of latency experienced over all requests, which is different from the count of write requests. Agreed that the naming is not great.

Reply:
If this is the total latency, which metric contains a request count?

(If there isn't one, I don't know what you can do with total latency if there isn't a count to divide it with.)

Reply:
Okay, that kind of makes sense. Can you fix the description for these metrics in the docs here?

(BTW it might be better to use "cumulative" or "delta" in the data type column here, instead of "sum" which doesn't mean anything in the context of GCM.)

Kind of unfortunately that read and write request counts are two separate metrics; normally I think I would want to see that modeled as a single metric with a label for read/write. But I assume we're also constrained there by the upstream receiver, since we don't want to apply transformations to metrics.

Reply:
Yeah, that's the currently behavior of the upstream receiver. We could certainly look at updating that in the future.

Descriptions fixed, sum updated to cumulative.

Reply:
I noticed that there's actually two different axes here. It feels like the request count should really be one metric with `operation={read,write,range_slice}` and `status={ok,timeout,unavailable}` labels, since right now there's just 3x3 metrics representing the cross product of those labels.

Reply:
+1 for the two proposed changes:

1. Change `50p/99p/max` to histogram with `{operation=read / write}` label. This seems to be a more long-term change since the support for histogram is not there yet.

2. Change
```
workload.googleapis.com/cassandra.client.request.read.latency.count
workload.googleapis.com/cassandra.client.request.read.timeout.count
workload.googleapis.com/cassandra.client.request.read.unavailable.count
workload.googleapis.com/cassandra.client.request.write.latency.count
workload.googleapis.com/cassandra.client.request.write.timeout.count
workload.googleapis.com/cassandra.client.request.write.unavailable.count
```

to something like

```
workload.googleapis.com/cassandra.client.request.count {operation=read / write, status=ok / timeout / unavailable}
```

This one seems like a more straight forward improvement. Related upstream PR: https://github.com/open-telemetry/opentelemetry-java-contrib/pull/43 (Handle list). While it could be done in the future, it does require a breaking change or we have to give customer options to switch between both metrics structures for backward compatibility. It might be worth waiting a bit instead of going though that hassle.

Reply:
Current status: waiting for upstream PR: https://github.com/open-telemetry/opentelemetry-java-contrib/pull/118 (
Update cassandra jmx metrics script to combine similar metrics into labelled metric).